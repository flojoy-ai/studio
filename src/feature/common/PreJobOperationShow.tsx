import { Modal, useMantineTheme } from "@mantine/core";
import { useMediaQuery } from "@mantine/hooks";
import { useEffect, useRef } from "react";

type PreJobOperationShowProps = {
  opened: boolean;
  outputs: string[];
  close: () => void;
};

const PreJobOperationShow = ({
  outputs,
  opened,
  close,
}: PreJobOperationShowProps) => {
  const isMobile = useMediaQuery("(max-width: 50em)");
  const theme = useMantineTheme();
  const lastElem = useRef<HTMLDivElement>(null);

  useEffect(() => {
    if (lastElem.current && lastElem.current.scrollIntoView) {
      lastElem.current.scrollIntoView({ behavior: "smooth" });
    }
  }, [lastElem.current]);

  return (
    <Modal
      opened={opened}
      onClose={close}
      title="Pre job operation"
      overlayProps={{
        color:
          theme.colorScheme === "dark"
            ? theme.colors.dark[9]
            : theme.colors.gray[2],
        opacity: 0.55,
        blur: 3,
      }}
      fullScreen={isMobile}
      size={"70%"}
    >
      {outputs.map((output, i) => (
        <div
          key={`${output}-${i}`}
          ref={i === outputs.length - 1 ? lastElem : null}
          style={{
            background: theme.colors.dark[9],
            color: output.toLowerCase().includes("error")
              ? theme.colors.red[8]
              : theme.colors.gray[0],
            padding: "10px",
            fontFamily: "monospace",
            whiteSpace: "break-spaces",
            overflow: "hidden",
          }}
        >
          {output}
        </div>
      ))}
    </Modal>
  );
};

export default PreJobOperationShow;
