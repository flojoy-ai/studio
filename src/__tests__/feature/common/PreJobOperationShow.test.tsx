import { render } from "@testing-library/react";
import PreJobOperationShow from "@src/feature/common/PreJobOperationShow";

jest.mock("@mantine/core", () => ({
  __esModule: true,
  default: jest.fn(({ children }) => <div>{children}</div>),
  Modal: jest.fn(({ children }) => <div>{children}</div>),
  useMantineTheme: jest.fn(() => ({
    colors: {
      dark: [, , , , , , , , , "#000"],
      gray: ["gray", , ,],
      red: [, , , , , , , , "red"],
    },
  })),
}));

describe("PreJobOperationShow", () => {
  it("should render correctly", () => {
    const outputs = ["Rendered this text!"];
    const { container, getByText } = render(
      <PreJobOperationShow opened={true} close={jest.fn} outputs={outputs} />
    );
    const elem = getByText(outputs[0]);
    expect(elem).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
