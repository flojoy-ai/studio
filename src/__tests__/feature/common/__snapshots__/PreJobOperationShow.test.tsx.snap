// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreJobOperationShow should render correctly 1`] = `
<div>
  <div>
    <div
      style="background: rgb(0, 0, 0); color: rgb(0, 0, 0); padding: 10px; font-family: monospace; white-space: break-spaces; overflow: hidden;"
    >
      Rendered this text!
    </div>
  </div>
</div>
`;
